source_code/backend/CS_department_monitor_backend/src/main/java/com/cs5324/monitorbackend/service/EventService.java

```diff
@@ -1,15 +1,67 @@
 package com.cs5324.monitorbackend.service;
 
+import com.cs5324.monitorbackend.entity.Event;
+import com.cs5324.monitorbackend.entity.Notification;
+import com.cs5324.monitorbackend.entity.User;
+import com.cs5324.monitorbackend.entity.enums.ItemStatus;
+import com.cs5324.monitorbackend.exception.EventDoesNotExistException;
 import com.cs5324.monitorbackend.repository.EventRepository;
+import com.cs5324.monitorbackend.repository.UserRepository;
 import jakarta.annotation.Resource;
 import lombok.RequiredArgsConstructor;
 import lombok.extern.slf4j.Slf4j;
 import org.springframework.stereotype.Service;
 
+import java.util.HashSet;
+import java.util.Set;
+import java.util.Optional;
+import java.util.UUID;
+
 @Slf4j
 @Service
 @RequiredArgsConstructor
 public class EventService{
     @Resource
     private final EventRepository eventRepo;
+
+    @Resource
+    private final UserRepository userRepo;
+
+    public Set<Event> getEvents(UUID userId){
+
+        Optional<User> user = userRepo.findById(userId);
+        Set<Event> allOwnedEvents = user.get().getEvents();
+
+        Set<Event> ownedEvents = new HashSet<Event>();
+        for(Event event : allOwnedEvents){
+            if(event.getApprovalStatus() == ItemStatus.APPROVED){
+                ownedEvents.add(event);
+            }
+        }
+
+        return ownedEvents;
+    }
+
+    public Optional<Event> getEvent(UUID eventId){
+
+        Optional<Event> event = eventRepo.findById(eventId);
+        if(event.isEmpty()){
+            log.warn("Event not found");
+            throw new EventDoesNotExistException();
+        }
+
+        return event;
+    }
+
+    public Notification submitEdits(Event editedEvent){
+        editedEvent.setApprovalStatus(ItemStatus.PENDING);
+        editedEvent.setDateOfEvent(editedEvent.getDateOfEvent());
+        editedEvent.setPage(editedEvent.getPage());
+
+        Notification editedNotif = new Notification();
+        editedEvent.setNotification(editedNotif);
+
+        return editedNotif;
+    }
```

Comment:
- What if the given event, passed in from the controller, does not have a valid id?
- What if either `editedEvent.dateOfEvent == null` or `editedEvent.page == null`, will we overwrite the existing data with `null`?
- The edited event should be saved to the database?